### DISCUSSION TITLE
dhcp plugin with hostname

### DISCUSSION THREAD

**Discussion by CosmicToast** [containers/podman#10479] [Q&A]
I have the following conflist:
```json
{
        "cniVersion": "0.4.0",
        "name": "dhcp",
        "plugins": [
                {
                        "type": "bridge",
                        "bridge": "mybridge",
                        "ipam": {"type": "dhcp"}
                }
        ]
}
```
I'd normally expect the dhcp daemon to pass along all the things a typical dhcp client would, including hostname (in this case, the names of the containers). That would allow my external network's dhcp server to resolve it via dns.
As it is, it's being marked as `*` (i.e no hostname given).
I know cni plugins (at least some of them) can know the container hostname, since that's how dnsname works.
Is this a conscious decision re: the cni plugin, or do I need to do something else?

Specifically: the goal is for ipam-dhcp to send the container's hostname to the dhcp server so it can be properly registered.
Network itself works fine from what I can tell, it's just not easily resolvable.

**Comment by mheon** [comment]
@mccv1r0 Any ideas here?

**Reply by mccv1r0** [reply]
There is an enhancement requested in cni dhcp plugin already: https://github.com/containernetworking/plugins/issues/435

**Comment by akhilles** [comment]
I believe this works now with the following IPAM config:

```json
"ipam": {
  "type": "dhcp",
  "provide": [
    {
      "option": "host-name",
      "fromArg": "K8S_POD_NAME"
    }
  ]
}
```

**Reply by CosmicToast** [reply]
I'm recovering from surgery right now and my old test env isn't available atm, but I'll try and test this within the next couple of weeks.
Thanks!

**Reply by akhilles** [reply]
I've been testing locally and I believe there's a bug. See https://github.com/containernetworking/plugins/pull/755.

**Comment by TrevorBenson** [comment]
When the backend is netavark the network configuration appears to be different with ipam_options, and driver:
```
     "ipam_options": {
          "driver": "dhcp"
     }
```
I tried setting:
```
     "ipam_options": {
          "driver": "dhcp",
          "provide": {
                  "option": "host-name",
                  "fromArg": "HOSTNAME"
          }
```
Which leads to **Error reading network config file**. Does this same functionality exist under netavark?

**Reply by Luap99** [reply]
No it does not, i.e. see https://github.com/containers/netavark/issues/676
